### PR TITLE
Fix Issue 933

### PR DESCRIPTION
**Pull request by h-kanazawa**
#### Which issue does this close?

Closes https://github.com/LearningLocker/enterprise/issues/933

#### Describe your changes providing screenshots of UI changes if necessary.

- Add new APIs
  - Delete user-organisations
  - Update user-organisationSettings
- Change Update User API (❗BREAKING CHANGE❗)
  - Non site-admins can change only name and own password)
  - Other fields are ignored
- Change Create User API (❗BREAKING CHANGE❗)
  - Non site-admins can set only name, email, isExpanded, organisations(own org)
  - Other fields are ignored

### CODE REVIEW

File: api/src/routes/userOrganisationSettings/router-test.js [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+    });
+
+    it('can update a its own organisationSettings of org1 if body has only updatable fields', async () => {
+      await apiApp
+        .patch(`/v2/users/${loginUser._id}/organisationSettings/${org1Id}`)
+        .set('Authorization', `Bearer ${nonUserManagerOrg1Token}`)
+        .send({ samlEnabled: false })
+        .expect(200);
+
+      // const updatedLoginUser = await User.findOne({ _id: loginUser._id });
```
Comment (ryansmith94) [inline api/src/routes/userOrganisationSettings/router-test.js]: Is this a TODO because it needs to be done in Enterprise instead @h-kanazawa?

Comment (h-kanazawa) [reply api/src/routes/userOrganisationSettings/router-test.js]: Yes, the assert is for enterprise.

**Comment by h-kanazawa** [comment]
Thanks @HT2-QA . I fixed the second issue

**Comment by h-kanazawa** [comment]
Fixed issue 03

**Pull request merged**